src/plugins/intel_gpu/src/graph/graph_optimizer/concat_input_order.cpp

```diff
@@ -48,7 +48,7 @@ bool can_shuffle_features(program_node& node, stream& stream) {
     if (pass_through) {
         // Primitives that are feature order invariant, pass-through shuffled features to users
         for (auto& user : node.get_users()) {
-            if (!can_shuffle_features(*user, stream))
+            if (!can_shuffle_features(*user, concat_node, stream))
```

Comment:
Please replace `concat_node` with `node`, since nodes on the next graph levels actually have concat's users in dependencies rather than concat node itself 
```suggestion
            if (!can_shuffle_features(*user, node, stream))
```

Reply:
Updated the code as you commented, really thanks for your comments.